runtime/interpreter/encode.go

```diff
@@ -793,8 +793,13 @@ func (e *Encoder) encodeArray(
 		return err
 	}
 
+	// Pre-allocate and reuse valuePath.
+	valuePath := append(path, "")
```

Comment:
`append` mutates the argument, so this would effect the calling code. The source argument should be a copy:

```suggestion
	valuePath := append(path[:], "")
```

This is still better than the original, which copied for each element

Reply:
Unless I'm mistaken, `valuePath := append(path,"")` followed by `valutePath[len(path)] = "abc"` won't mutate `path` in our particular case.  More details at https://github.com/onflow/cadence/pull/858#issuecomment-829652459

Reply:
Resolving as per discussion below